crates/ruff_python_formatter/resources/test/fixtures/ruff/expression/dict.py

```diff
@@ -0,0 +1,26 @@
+# before
+{  # open
+ key# key
+ : # colon
+ value# value
+} # close
+# after
+
+
+{**d}
```

Comment:
Could you add a case for comments and `**`? something like
```python
{**a, #  leading
** # middle
b # trailing
}
```

Reply:
good idea

note that black (even for the first existing example) combines the comments more aggressively than i managed in this pr

me:

```
{
    # open
    key: value,  # key  # colon  # value
}  # close
```

black:

```
{key: value}  # open  # key  # colon  # value  # close
```

i couldn't figure out how to prevent e.g. the `# open` from being rendered when i printed the opening bracket

Reply:
ah, interesting. your example formats as 

```
{
    **a,  #  leading  # middle
    **b,  # trailing
}
```

which seem wrong (i would have attributed middle to b, not a.). is that a formatting bug in my pr or the parser assigning it to the "wrong" value?

Reply:
you need to manually override this in `place_comments`. I tried to write some better docs for dangling comment in https://github.com/astral-sh/ruff/pull/5170, but switching leading and trailing is also done in `place_comments`.

Reply:
actually, was this a reply to ` # open  # key  # colon  # value  # close` or `# leading  # middle\n # trailing` ?

(and if different for the two, what about the other one)

Reply:
The formatting of `key: value` comments looks right. We don't currently match black for a bunch of comment placement and i don't know yet if we can sustain those deviations, but i want to go forward with this design.

For the `**b` case, you have to add a function to `place_comments` that uses the `SimpleTokenizer` or one of the wrapper functions to find `**` and assigns comments to nodes depending on whether they are left or right of the `**`. Searching `placements.rs` for "token" will show some reference code.

Reply:
ok i got something working i think